pydy/tests/test_system.py

```diff
@@ -395,3 +397,27 @@ def test_integrate(self):
         sys = System(self.kane, times=times)
         with testing.assert_raises(NotImplementedError):
             sys.generate_ode_function(generator='made-up')
+
+
+class TestJointSystem():
+
+    def setup(self):
+        self.system = System.using_joints()
```

Comment:
http://stackoverflow.com/questions/19305296/multiple-constructors-in-python-using-inheritance
Its my reference

> A simple example from the standard library is datetime.datetime, which can be constructed with now, fromtimestamp, or a few other alternate constructors, besides the default.

Also, I want to keep our present classes intact as long as we have options. Wanted to get your views on this.